### DISCUSSION TITLE
React Hooks with Material UI and Validation (via Controller)

### DISCUSSION THREAD

**Discussion by btoro** [react-hook-form/react-hook-form#2342] [Q&A]
Using a controller, I managed to setup all the proper material UI components. Things work well when I dont make use of rules or watch functions. However, when I apply any rules, I start to have weird issues pop up. 

In the gif bellow I enabled the required rule on the controller. When I try to select an option at the beginning, everything works smoothly. If I try to go ahead without selecting an option, the error message pops up. This is correct. Now, when I go to select an option, I am unable to. The error message clears, but the radio element is never selected.

Any ideas what is going on? I see no errors / warnings in the console.

I have a similar issue if I use a watch on an input. I lose the ability to alter its value.

![screencast-localhost_3000-2020 07 23-14_14_28](https://user-images.githubusercontent.com/15056753/88334687-66abb300-ccef-11ea-8d0d-e7a4bd6a54dc.gif)

```        
<div>
          <Controller
            as={
              <RadioGroup aria-label={itemData.questionId.toString()}>
                {itemData.answerOption.map((item, index) => (
                  <FormControlLabel
                    value={"a" + item.answerId.toString()}
                    control={<Radio />}
                    label={item.display}
                    key={index}
                  />
                ))}
              </RadioGroup>
            }
            name={itemData.questionId.toString()}
            control={control}
            defaultValue={
              answerValues[itemData.questionId]
                ? _.keys(answerValues[itemData.questionId])[0]
                : ""
            }
            rules={{ required: "Please select an option." }}
          />
          <ErrorMessage errors={errors} name={itemData.questionId.toString()} />
        </div>
```

You will notice that I try to pull defaultValue from a stored array. Even If i disable that behavior (defaultValue=""), I still get this issue.

**Comment by bluebill1049** [comment]
can you provide an isolated codesandbox for this issue?

**Reply by btoro** [reply]
https://codesandbox.io/s/validaionerror-opwj5?file=/src/App.js

FYI, my goal is to create forms dynamically from json questionnaires. 

To test, try to hit next. you will see the error message. 
Then try to select a radio button. you wont be able to.

Thanks!

**Reply by btoro** [reply]
@bluebill1049 Any luck with this?

**Reply by bluebill1049** [reply]
I think your defaultValue looks suspicious, try to pay attention to the console.log see if you get any warning as well. i wouldn't recommend to do dynamic defaultValue, if you want to change value use reset or setValue.

**Reply by btoro** [reply]
I set defaultValue to "" and I still see the same issue. See the updated codebox.

**Reply by bluebill1049** [reply]
then perhaps set up a codesandbox isolate that radio group, then you can identify if there is any side effect which cause this issue.

**Comment by cliffordp** [comment]
Could this be solved by https://github.com/react-hook-form/react-hook-form/discussions/2281 ?

**Reply by btoro** [reply]
Im not sure how that relates, or how the setTimeout callback relates here
